tests/net/virtual/src/main.c

```diff
@@ -1004,21 +1004,24 @@ static void test_virtual_recv_data_from_tunnel_fail(void)
 	test_virtual_recv_data_from_tunnel(3, false);
 }
 
-void test_main(void)
+static void *setup(void)
 {
-	ztest_test_suite(net_virtual_test,
-			 ztest_unit_test(test_virtual_setup),
-			 ztest_unit_test(test_address_setup),
-			 ztest_unit_test(test_virtual_attach_and_detach),
-			 ztest_unit_test(test_virtual_set_mtu),
-			 ztest_unit_test(test_virtual_get_mtu),
-			 ztest_unit_test(test_virtual_set_peer),
-			 ztest_unit_test(test_virtual_get_peer),
-			 ztest_unit_test(test_virtual_verify_name),
-			 ztest_unit_test(test_virtual_send_data_to_tunnel),
-			 ztest_unit_test(test_virtual_recv_data_from_tunnel_ok),
-			 ztest_unit_test(test_virtual_recv_data_from_tunnel_fail)
-			 );
-
-	ztest_run_test_suite(net_virtual_test);
+	test_virtual_setup();
+	test_address_setup();
+	return NULL;
 }
+
+ZTEST(virtual_test_suite, test_virtual)
+{
+	test_virtual_attach_and_detach();
+	test_virtual_set_mtu();
+	test_virtual_get_mtu();
+	test_virtual_set_peer();
+	test_virtual_get_peer();
+	test_virtual_verify_name();
+	test_virtual_send_data_to_tunnel();
+	test_virtual_recv_data_from_tunnel_ok();
+	test_virtual_recv_data_from_tunnel_fail();
```

Comment:
Do all of them have a dependency problem? Could we define it by ZTEST macro as possible?

Reply:
@enjiamai 
As the ZTEST runs the test cases alphabetically, the running sequence is different from the old APIs.
Many test cases under tests/net have to be run in specific order, there are two ways to handle it:
1. set the test case name to test_[domain]_[number]_[test case name], this PR is an example
    Advantage: easy to move the test cases to new API, when 1 test case failed it is easy to locate the function.
    Disadvantage: when adding a new test case, the [number] of other test cases in the test suite may also be changed
2. use a function to test the cases in good order. tests/net/vlan is an example
    Advantage: the dependency is clear
    Disadvantage: if 1 test case fail, many functions may be checked. Takes time to find out the dependency.
Which one do you prefer? We need to find out an approach and then modify all the other tests.

Reply:
Ok, thanks for the clarification.